cpp/src/arrow/flight/CMakeLists.txt

```diff
@@ -26,6 +26,9 @@ endif()
 if(WIN32)
   list(APPEND ARROW_FLIGHT_LINK_LIBS ws2_32.lib)
 endif()
+if(ARROW_USE_OPENSSL)
```

Comment:
`if(APPLE AND ARROW_USE_OPENSSL)` may be better because this is just needed on macOS.